eng/scripts/CodeCheck.ps1

```diff
@@ -0,0 +1,69 @@
+#requires -version 5
+<#
+.SYNOPSIS
+This script runs a quick check for common errors, such as checking that Visual Studio solutions are up to date or that generated code has been committed to source.
+#>
+param(
+    [switch]$ci
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 1
+Import-Module -Scope Local -Force "$PSScriptRoot/common.psm1"
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../../"
+
+[string[]] $errors = @()
+
+try {
+    #
+    # Solutions
+    #
+
+    Write-Host "Checking that solutions are up to date"
+
+    Get-ChildItem "$repoRoot/*.sln" -Recurse | % {
+        Write-Host "  Checking $(Split-Path -Leaf $_)"
+        $slnDir = Split-Path -Parent $_
+        $sln = $_
+        & dotnet sln $_ list `
+            | ? { $_ -ne 'Project(s)' -and $_ -ne '----------' } `
+            | % {
+                $proj = Join-Path $slnDir $_
+                if (-not (Test-Path $proj)) {
+                    $errors += "Missing project. Solution references a project which does not exist: $proj. [$sln] "
+                }
+            }
+        }
+
+    #
+    # Generated code check
```

Comment:
Good idea 👍

Reply:
This is just a start. I'm hoping to add more checks in the future, like updating resx files, generated baselines, and maybe even a C# code formatter.